### DISCUSSION TITLE
cockpit frontend run

### DISCUSSION THREAD

**Discussion by jenetjoshicnit** [cockpit-project/cockpit#20849] [Q&A] [answered]
Here in this project, i want to run the frontend of this project, after cloning this github repo,i tried using the commands `npm start` , `npm run dev` and `npm run build` , but theese scripts are also not there in the package.json also, so i need to get an outline of how to run the frontend code of cockpit. Anyoone cracked running the frontend of cockpit , please do help me.

**Comment by subhoghoshX** [comment]
Hey @jenetjoshicnit, thanks for your interest! You can find detailed instructions on setting up the development environment and running the project at https://github.com/cockpit-project/cockpit/blob/main/HACKING.md.

**Reply by jenetjoshicnit** [reply]
hi, @subhoghoshX ,thank you for your reply, i opened the link you provided, but iam working in windows and the commands mentioned in the doc is for linux users , but i want to open the frontend in my windows, Here, in the package.json file there is no start server and dev server so i need open my frontend in my local. can you please help me with this, if you have done any.

**Reply by subhoghoshX** [reply, marked as answer]
To develop Cockpit, you'll need a Linux installation. The Cockpit frontend serves as the graphical interface for a Linux server (not for Windows or macOS), so you can't run the frontend independently without a Linux server. However, once the server is set up, it can be accessed through a browser from any operating system, including Windows.

You might be able to follow the instructions in [HACKING.md](https://github.com/cockpit-project/cockpit/blob/main/HACKING.md) using Windows Subsystem for Linux (WSL), but this is not officially supported or tested on as far as I know.

**Reply by jenetjoshicnit** [reply]
ok, thank you @subhoghoshX for having this discussion with me but , i had one more query to solve regarding custom package , can you please share with me the concept of how we create a custom package of cockpit module after the modification development we did. However, thanks for the support that you made so far

**Reply by martinpitt** [reply]
@jenetjoshicnit *please* read HACKING.md -- it's all there. E.g. https://github.com/cockpit-project/cockpit/blob/main/HACKING.md#installation-from-upstream-sources explains how to install from upstream or make debs/rpms. If you modified a single page, you can also distribute e.g. `dist/systemd` (or whichever other page you modified) and install it into `/usr/local/share/cockpit/`.

**Reply by jenetjoshicnit** [reply]
hi, @martinpitt , thank you for the reply, I referred this documents on cockpit, here my requirement is to change some of the UI elements in cockpit ,but according to the document, when I change some and take build using these following commands,

--> mkdir -p ~/.local/share/
--> ln -s $(pwd)/dist ~/.local/share/cockpit
--> ./build.js -w

I followed these steps one by one , to see the change in the build which I take after I change the UI elements, but it is not reflecting ,what can I do to change my UI elements through code and see the changes reflecting, @subhoghoshX and @martinpitt , if any of you have contributed to the UI changes of cockpit interface. Please let me know how it is done. I have tried many ways doing it.
